rs/sns/integration_tests/src/get_metrics.rs

```diff
@@ -0,0 +1,203 @@
+    let alice = get_account(ALICE, 0);
+    let bob = get_account(BOB, 0);
+    const NOW: u64 = 1_748_854_120;
+
+    // Prepare the world:
+    // 1. create the ledger canister and add two accounts
+    // for ALICE and BOB with INITIAL_BALANCE.
+    let ledger_canister_id = {
+        let mut initial_balances = HashMap::new();
+        add_balance(&mut initial_balances, alice, INITIAL_BALANCE);
+        add_balance(&mut initial_balances, bob, INITIAL_BALANCE);
+        install_ledger(&state_machine, initial_balances)
+    };
+
+    // 2. make transfers to create ledger blocks.
+    {
+        const TIME_STEP: u64 = 10;
+        const NUM_TRANSACTIONS: u64 = 10;
+        const TRANSFER_AMOUNT: u64 = 1_000_000;
+        for tx in 0..NUM_TRANSACTIONS {
+            let created_at_time = TimeStamp::new(NOW + tx * TIME_STEP, 0);
+
+            let _ = icrc1_transfer(
+                &state_machine,
+                ledger_canister_id,
+                alice,
+                bob,
+                TRANSFER_AMOUNT,
+                Some(created_at_time),
+                Some(FEE),
+                Some(vec![]),
+            );
+        }
+    }
+
+    // 3. start the governance canister and pass the
+    // already initialised ledger canister to it.
+    let governance_canister_id = {
+        let wasm = build_governance_sns_wasm().wasm;
+        let governance = GovernanceCanisterInitPayloadBuilder::new()
+            .with_root_canister_id(PrincipalId::new_anonymous())
+            .with_swap_canister_id(PrincipalId::new_anonymous())
+            .with_ledger_canister_id(ledger_canister_id.into())
+            .build();
+
+        let args = Encode!(&governance).unwrap();
+        state_machine
+            .install_canister(wasm.clone(), args, None)
+            .unwrap()
+    };
+
+    {
+        // Prepare the payload to get metrics during the last month.
+        let time_window_seconds = 30 * 24 * 3600; // 1 month.
+        let payload = GetMetricsRequest {
+            time_window_seconds: Some(time_window_seconds),
+        };
+
+        let Ok(observed_result) = try_get_metrics(&state_machine, governance_canister_id, payload)
+        else {
+            panic!("Received an Error upon fetching the metrics")
+        };
+
+        let Some(get_metrics_result) = observed_result.get_metrics_result else {
+            panic!("Expected a non-empty response");
+        };
+
+        let get_metrics_response::GetMetricsResult::Ok(metrics) = get_metrics_result else {
+            panic!("Expected to get an Ok() from the response");
+        };
+
+        {
+            let Some(num_recently_submitted_proposals) = metrics.num_recently_submitted_proposals
+            else {
+                panic!("Expected `num_recently_submitted_proposals` to be Some(0)");
+            };
+
+            assert_eq!(
+                num_recently_submitted_proposals, 0,
+                "Expected 0 proposals to be submitted, got {}",
```

Comment:
I think we should set up the test to have at least one proposal, so the tests covers all the functionality together (including the proposal counting).

Reply:
Actually, let's have one proposal that is IN the requested window and one that is OUTSIDE. Wdyt?